src/main/scala/mercury/Controller.scala

```diff
@@ -36,19 +36,28 @@ class Controller extends unfiltered.filter.Plan {
 
       ResponseString(html.history.render(url.getOrElse(""), history).body) ~> HtmlContent
 
-    case GET(Path("/history.json") & Params(p)) =>
+    case r@ GET(Path("/history.json") & Params(p)) =>
       val url = p("url").headOption getOrElse sys.error("missing url")
       val callback = p("callback").headOption
+      val tz = p("tz").headOption
       val history = Store.findHistory(url)
 
-      val json = renderJsonResponse(history)
+      val json = renderJsonResponse(history, tz)
 
-      callback.map(
+      val resp = callback.map(
         c => ResponseString(s"$c($json)") ~> JsContent
       ) getOrElse (
         ResponseString(json) ~> JsonContent
       )
 
+      val origin = r.headers("Origin").toList.headOption
+
+      Cors.headers(origin).map { originHeader =>
+        Ok ~> originHeader ~> resp
+      }.getOrElse {
+        Ok ~> resp
+      }
+
```

Comment:
can we not just set a cors header allowing access from *? since (I think) that's the effect of this code anyway.